opm/simulators/wells/ParallelPAvgDynamicSourceData.cpp

```diff
@@ -148,7 +148,19 @@ void Opm::ParallelPAvgDynamicSourceData::defineCommunication()
 
     // 4) Build translation mapping from source term element indices to
     //    storage indices.
-    this->storageIndex_.resize(allIndices.size());
+    //
+    //    Note that if the source terms aren't all active--e.g., if a well
+    //    is connected in deactivated cells--then allIndices is not a
+    //    permutation of 0..allIndices.size()-1 and the maximum source
+    //    location may exceed size()-1.  Resize the storageIndex_ according
+    //    to the largest source location ID.
+    if (auto maxIxPos = std::max_element(allIndices.begin(), allIndices.end());
+        maxIxPos != allIndices.end())
```

Comment:
maybe a bit wordy guard against an empty vector.

Reply:
> maybe a bit wordy guard against an empty vector.

I'm not entirely sure I understand that comment.  Yes, this guards against an empty vector, but I also need the maximum value (`*maxIxPos`) in the `.resize()` call below.

Reply:
yeah, i just would personally have structured it a bit differently with an if (!empty)  around the resize nest. no need to change anything.